library/src/reduction/reduction_types.hpp

```diff
@@ -41,50 +41,14 @@ namespace hiptensor
                           || (opReduce == HIPTENSOR_OP_MIN) || (opReduce == HIPTENSOR_OP_MAX),
                       "opReduce is not supported");
 
-        constexpr auto reduceOpId = (opReduce == HIPTENSOR_OP_ADD)   ? ck::ReduceTensorOp::ADD
-                                    : (opReduce == HIPTENSOR_OP_MUL) ? ck::ReduceTensorOp::MUL
-                                    : (opReduce == HIPTENSOR_OP_MIN) ? ck::ReduceTensorOp::MIN
-                                                                     : ck::ReduceTensorOp::MAX;
+        constexpr auto reduceOpId = (opReduce == HIPTENSOR_OP_ADD)
```

Comment:
To make this more extensible, you might want to throw it into a template class with overrides for each specific op mapping.

```
template<hiptensorOperator_t>
struct CKOperator;

template<>
struct CKOperator<HIPTENSOR_OP_ADD>
{ 
  static constexpr inline auto value() { return ck::ReduceTensorOp::ADD; }
};
...
```